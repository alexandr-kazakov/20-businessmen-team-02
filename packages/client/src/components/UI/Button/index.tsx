import React, { ButtonHTMLAttributes, memo, FC } from 'react'
import classnames from 'classnames'
import { ButtonStyles } from './types'
import styles from './styles.module.scss'

type OwnProps = {
  variant: ButtonStyles
} & ButtonHTMLAttributes<HTMLButtonElement>

type Props = OwnProps

const Button: FC<Props> = props => {
  const { variant, children, ...buttonProps } = props

  const className = classnames(styles.button, {
    [styles.button_primary]: variant === ButtonStyles.primary,
    [styles.button_secondary]: variant === ButtonStyles.secondary,
  })

  return (
    <button {...buttonProps} className={className}>
      {children}
    </button>
  )
}

export default memo(Button)
