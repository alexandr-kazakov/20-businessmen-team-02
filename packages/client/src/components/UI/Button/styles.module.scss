.button {
  background: none;
  border: none;
  padding: 10px;
  color: #fff;
  cursor: pointer;

  &.primary {
    width: 100%;
    background-color: #d9534f;
    border-radius: 10px;
  }

  &.secondary {
    background-color: transparent;
  }

  &:disabled {
    opacity: 0.7;
  }
}
