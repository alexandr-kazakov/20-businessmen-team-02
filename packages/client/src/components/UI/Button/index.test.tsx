import React from 'react'
import { render } from '@testing-library/react'
import { Button } from '@/components/UI/Button/index'

test('should render Button component', () => {
  const { asFragment } = render(<Button>BUTTON</Button>)

  expect(asFragment()).toMatchSnapshot()
})
