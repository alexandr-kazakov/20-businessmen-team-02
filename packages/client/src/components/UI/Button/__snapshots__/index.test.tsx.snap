// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button component 1`] = `
<DocumentFragment>
  <button
    class=""
  >
    BUTTON
  </button>
</DocumentFragment>
`;
